Mlem/App/Views/Shared/Labels/FullyQualifiedLabelView.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  FullyQualifiedLabelView.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-05-19.
+//
+
+import Foundation
+import MlemMiddleware
+import SwiftUI
+
+enum FullyQualifiedLabelStyle {
+    case small(showAvatar: Bool)
+    case medium(showAvatar: Bool)
+    case large(showAvatar: Bool)
+    
+    var avatarSize: CGFloat {
+        switch self {
+        case .small: AppConstants.smallAvatarSize
+        case .medium: AppConstants.mediumAvatarSize
+        case .large: AppConstants.largeAvatarSize
+        }
+    }
+    
+    var showAvatar: Bool {
+        switch self {
+        case let .small(showAvatar): showAvatar
+        case let .medium(showAvatar): showAvatar
+        case let .large(showAvatar): showAvatar
+        }
+    }
+    
+    var instanceLocation: InstanceLocation {
+        switch self {
+        case .small: .trailing
+        case .medium: .trailing
+        case .large: .bottom
+        }
+    }
+}
+
+/// View for rendering fully qualified labels (i.e., user or community names)
+struct FullyQualifiedLabelView: View {
+    // these aren't used now but they will be for flairs
+    @Environment(\.postContext) var postContext: (any Post1Providing)?
+    @Environment(\.communityContext) var communityContext: (any Community3Providing)?
```

Comment:
The value of this needs to be set somewhere - it isn't yet. The post view expect `communityContext` to be the context of the community you're viewing, whereas `FullyQualifiedLabelView` expects it to be the community of the post it's attached to. So, `communityContext` needs to be set on the label views from the post views.

Reply:
The design is for `communityContext` to be set:
- By the expanded post view
- By the community feed view
In the feed, there is no community context, and so it is `nil`.

> The post views expect communityContext to be the context of the community you're viewing, whereas FullyQualifiedLabelView expects it to be the community of the post it's attached to.

Outside of a specific community or expanded post, `communityContext` isn't useful to have defined for the label--afaik it's only used to determine whether the user should be flaired as moderator, which can only be determined using a full `Community3Providing`. Since we don't want to upgrade every single community we fetch, it therefore doesn't make much sense to provide this to the label unless we're in a detailed context (expanded post or community feed); in those contexts, the current community and the post's community are identical.

Once we drop 0.18 support, we can get rid of this altogether and just rely on the post context.

Reply:
Ah, that makes sense 👍